cluster-autoscaler/core/scale_up_test.go

```diff
@@ -937,6 +937,42 @@ func TestScaleUpBalanceAutoprovisionedNodeGroups(t *testing.T) {
 	assert.True(t, expandedGroupMap["autoprovisioned-T1-2-1"])
 }
 
+func TestScaleUpToMeetNodeGroupMinSize(t *testing.T) {
+	podLister := kube_util.NewTestPodLister([]*apiv1.Pod{})
+	listers := kube_util.NewListerRegistry(nil, nil, podLister, nil, nil, nil, nil, nil, nil, nil)
+	provider := testprovider.NewTestCloudProvider(func(nodeGroup string, increase int) error {
+		assert.Equal(t, "ng1", nodeGroup)
+		assert.Equal(t, 1, increase)
+		return nil
+	}, nil)
+
+	n1 := BuildTestNode("n1", 100, 1000)
+	n2 := BuildTestNode("n2", 100, 1000)
+	provider.AddNodeGroup("ng1", 2, 10, 1)
+	provider.AddNode("ng1", n1)
+	provider.AddNodeGroup("ng2", 1, 10, 1)
+	provider.AddNode("ng2", n2)
+
+	options := config.AutoscalingOptions{
+		EstimatorName:  estimator.BinpackingEstimatorName,
+		MaxCoresTotal:  config.DefaultMaxClusterCores,
+		MaxMemoryTotal: config.DefaultMaxClusterMemory,
+	}
+	context, err := NewScaleTestAutoscalingContext(options, &fake.Clientset{}, listers, provider, nil, nil)
+	assert.NoError(t, err)
+
+	nodes := []*apiv1.Node{n1, n2}
+	nodeInfos, _ := nodeinfosprovider.NewDefaultTemplateNodeInfoProvider(nil).Process(&context, nodes, []*appsv1.DaemonSet{}, nil, time.Now())
+	clusterState := clusterstate.NewClusterStateRegistry(provider, clusterstate.ClusterStateRegistryConfig{}, context.LogRecorder, NewBackoff())
+	clusterState.UpdateNodes(nodes, nodeInfos, time.Now())
+
+	scaleUpStatus, err := ScaleUpToNodeGroupMinSize(&context, clusterState, nodeInfos)
+	assert.NoError(t, err)
+	assert.True(t, scaleUpStatus.WasSuccessful())
+	assert.Equal(t, 1, len(scaleUpStatus.ScaleUpInfos))
+	assert.Equal(t, 2, scaleUpStatus.ScaleUpInfos[0].NewSize)
```

Comment:
nit: Worth also asserting it was `ng1`, not `ng2` that got scaled up.

Reply:
Actually the node group id check was added in line 944. Will add another assert here to be clearer.